libkbfs/journal_block_server_test.go

```diff
@@ -15,24 +15,29 @@ import (
 )
 
 func setupJournalBlockServerTest(t *testing.T) (
-	tempdir string, config Config, jServer *JournalServer) {
+	tempdir string, config *ConfigLocal, jServer *JournalServer) {
+	config = MakeTestConfigOrBust(t, "test_user")
+
 	tempdir, err := ioutil.TempDir(os.TempDir(), "journal_block_server")
 	require.NoError(t, err)
-	config = MakeTestConfigOrBust(t, "test_user")
-	log := config.MakeLogger("")
-	jServer = makeJournalServer(
-		config, log, tempdir, config.BlockCache(), config.DirtyBlockCache(),
-		config.BlockServer(), config.MDOps(), nil, nil)
-	ctx := context.Background()
-	err = jServer.EnableExistingJournals(
-		ctx, TLFJournalBackgroundWorkPaused)
+	// Clean up the tempdir if anything in the setup fails/panics.
+	defer func() {
+		if r := recover(); r != nil {
+			CheckConfigAndShutdown(t, config)
```

Comment:
Maybe you could check `if recover()` in the teardown function, to decide whether to require the error is nil?